src/snowflake/cli/plugins/nativeapp/manager.py

```diff
@@ -98,7 +99,30 @@ def process(self, *args, **kwargs):
         pass
 
 
-class NativeAppManager(SqlExecutionMixin):
+class FindObjectRowMixin:
+    def __init__(self):
+        if not isinstance(self, SqlExecutionMixin):
+            raise TypeError(
+                "FindObjectRowMixin can only be used with classes that also mix in SqlExecutionMixin."
```

Comment:
I will say I'm not entirely sure the best place to put this mixin. We could just include the implemented function (find_row_by_object_name) in SqlExecutionMixin itself as well though I'm not sure if the use cases justify it.